src/test/java/edu/ksu/canvas/attendance/submitter/AssignmentSubmitterUTest.java

```diff
@@ -195,13 +240,24 @@ public void setup() {
         Whitebox.setInternalState(assignmentSubmitter, "attendanceService", attendanceService);
         Whitebox.setInternalState(assignmentSubmitter, "canvasApiWrapperService", canvasApiWrapperService);
         Whitebox.setInternalState(assignmentSubmitter, "assignmentValidator", assignmentValidator);
+        Whitebox.setInternalState(assignmentSubmitter, "studentRepository", studentRepository);
+        Whitebox.setInternalState(assignmentSubmitter, "studentService", studentService);
 
 
 
     }
 
     @Test
     public void submitCourseAttendancesHappyPath() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS00112)